src/main/java/org/broadinstitute/barclay/argparser/CommandLinePluginDescriptor.java

```diff
@@ -76,103 +107,126 @@
-     * Implementations of this method should minimally validate that all of values that have
+     * Implementations of this method should minimally validate that all values that have
      * been specified on the command line have a corresponding plugin instance (this will
      * detect a user-specified value for which there is no corresponding plugin class).
      *
-     * @throws CommandLineException if a plugin value has been specified that
-     * has no corresponding plugin instance (i.e., the plugin class corresponding to the name
-     * was not discovered)
+     * @throws CommandLineException if a plugin value has been specified that has no corresponding
+     * plugin instance (i.e., the plugin class corresponding to the name was not discovered)
+     * or the plugins otherwise cannot be resolved per the descriptor's argument resolution policy.
      */
-    public abstract void validateArguments() throws CommandLineException;
+    public abstract void validateAndResolvePlugins() throws CommandLineException;
 
     /**
-     * @return the default plugins used for this instance of this descriptor as a list of Object. Used for
-     * help/doc generation.
+     * @return the default plugins enabled for this descriptor as a list of Object. Used for
+     * help/documentation generation for this descriptor.
+     *
+     * NOTE: calling this method after argument parsing (and thus after {@link #validateAndResolvePlugins}
+     * is called) may return a different list than calling it before parsing, since the resolution
+     * policy is implementation-dependent and may depend on the actual argument specified by the user
+     * on the command line.
      */
     public abstract List<Object> getDefaultInstances();
 
     /**
-     * @return an ordered List of actual plugin instances that have been specified on the command
-     * line, in the same order they were obtained/created by {@line #getInstance}).
+     * @return an ordered list containing the plugins that result from resolving all command line
+     * arguments with any default plugins that have been provided to this descriptor. This list
+     * represents the plugins that will actually be used by the consumer. The resolution policy is
+     * descriptor-dependent.
+     *
+     * NOTE: calling this method after argument parsing (and thus after {@link #validateAndResolvePlugins}
+     * is called) may return a different list than calling it before parsing, since the resolution
+     * policy is implementation-dependent and may depend on the actual argument specified by the user
+     * on the command line.
+     */
+    public abstract List<T> getResolvedInstances();
+
+    /**
+     * Return the allowable values for the String argument of this plugin descriptor
+     * that is specified by {@code longArgName}. Used for help/documentation generation for this descriptor.
+     . If the value is unrecognized
+     * the implementation should throw
+     * IllegalArgumentException.
```

Comment:
Nitpicking comment: this lines are weirdly formatted.

Reply:
Yeah. Fixed.